internal/controller/provisioner/objects.go

```diff
@@ -780,6 +780,11 @@ func (p *NginxProvisioner) buildNginxPodTemplateSpec(
 				container.Command = append(container.Command, "/agent/entrypoint.sh")
 				container.Args = append(container.Args, "debug")
 			}
+
+			if containerSpec.HostPort != nil && containerSpec.HostPort.Enable && len(container.Ports) > 0 {
+				container.Ports[0].HostPort = containerSpec.HostPort.Port
+			}
```

Comment:
The issue with this approach is that the nginx container is going to be exposed on X number of ports, which are directly correlated to the listener ports in the Gateway object.

You've only exposed a single hostPort, and are binding it to whatever the first container port is. When in reality, each container port should have its own hostPort. If you take a look at how we added NodePorts in the `values.yaml` file and then how we use that to set Service ports; it should be a similar structure, where each port maps to a Gateway listener port.